lib/xmerl/src/xmerl_lib.erl

```diff
@@ -45,6 +45,9 @@
 -include("xmerl_xsd.hrl").
 
 
+export_cdata(T) ->
+    ["<![CDATA[", T, "]]>"].
```

Comment:
Please keep in mind that `T` itself may contain `]]>`, in which case the result will be invalid, or rather, not what you would expect. One way to escape `]]>` in CDATA that I know of is to replace it with `]]]]><[CDATA[>`.

Reply:
Thanks for pointing that out. I suppose the only way to cater for that would be
```diff
diff --git a/lib/xmerl/src/xmerl_lib.erl b/lib/xmerl/src/xmerl_lib.erl
index 2e233637d6..3235b89af7 100644
--- a/lib/xmerl/src/xmerl_lib.erl
+++ b/lib/xmerl/src/xmerl_lib.erl
@@ -46,7 +46,13 @@
 
 
 export_cdata(T) ->
-    ["<![CDATA[", T, "]]>"].
+    ["<![CDATA[",
+     binary:replace(
+       iolist_to_binary(T),
+       <<"]]>">>,
+       <<"]]]]><![CDATA[>">>,
+       [global]),
+     "]]>"].
 
 %% Escape special characters `<' and `&', flattening the text.
 %% Also escapes `>', just for symmetry.
```

Reply:
Hm... I'm pretty sure this doesn't do the right thing.

OTOH, if the argument list `T` contains unicode data, eg `"ħ"` (same as `[295]`), `iolist_to_binary` will fail.

OTOH, if `T` contains something like `"Ä§"` (same as `[196, 167]`), `iolist_to_binary` will accept and convert it to `<<196, 167>>`, but `unicode:characters_to_list` will turn that into `"ħ"` (`[295]`).

Reply:
So looking at `xmerl_lib:export_text/1` it is not unicode safe at all.

The safest way I can see to handle this using existing code is to just convert `T` to graphemes using `string:to_graphemes/1` beforehand. (results using this below)

```erlang
37> io:format("~ts~n", [xmerl_lib:export_cdata(["some ", [<<"Ä§"/utf8>>, " ", "Ä§", " ħaa", <<" ❤️ "/utf8>>], " ]]> ", " other "])]).
some Ä§ Ä§ ħaa ❤️  ]]]]><![CDATA[>  other 
ok
38> io:format("~ts~n", [xmerl_lib:export_text(["some ", [<<"Ä§"/utf8>>, " ", "Ä§", " ħaa", <<" ❤️ "/utf8>>], " ]]> ", " other "])]). 
some ÃÂ§ Ä§ ħaa â¤ï¸  ]]&gt;  other 
```

I'm guess making `xmerl_lib:export_text/1` unicode safe should probably be done in another PR